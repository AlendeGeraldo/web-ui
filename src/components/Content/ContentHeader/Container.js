// CORE LIBS
import React from 'react';
import { useHistory } from 'react-router-dom';

// COMPONENTS
import ContentHeader from './index';

/**
 * Content Header Container.
 * This component is responsible for create a logic container for content header
 * with route control.
 */
const ContentHeaderContainer = ({ title, editable }) => {
  // getting history
  const history = useHistory();

  // HANDLERS
  const goBackHandler = () => history.goBack();

  // RENDER
  return (
    <ContentHeader
      title={title}
      editable={editable}
      handleGoBack={goBackHandler}
    />
  );
};

// EXPORT
export default ContentHeaderContainer;
