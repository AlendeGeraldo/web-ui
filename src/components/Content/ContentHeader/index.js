// CORE LIBS
import React from 'react';
import PropTypes from 'prop-types';

// UI LIBS
import { PageHeader } from 'antd';

// COMPONENTS
import EditableTitle from '../EditableTitle';

// STYLES
import './style.scss';

// editable title class name const
const editableTitleClassName =
  'ant-page-header-heading-title autosize-input-custom';

/**
 * Content Header.
 * This component is responsible for displaying the content header with editable
 * title and go back arrow.
 */
const ContentHeader = ({ handleGoBack, editable, title }) => {
  // COMPONENTS RENDERS
  // common title
  const renderTitle = () => title;
  // editable title
  const renderEditableTitle = () => (
    // editable title component
    <EditableTitle
      loading={false}
      title={title}
      handleSubmit={(titleSubmited) => alert(titleSubmited)}
      className={editableTitleClassName}
      editingClassName={`${editableTitleClassName} edit-mode`}
    />
  );

  // RENDER
  return (
    // page header component
    <PageHeader
      className='contentHeader'
      title={editable ? renderEditableTitle() : renderTitle()}
      onBack={handleGoBack}
    />
  );
};

// PROP TYPES
ContentHeader.propTypes = {
  /** content header go back function */
  handleGoBack: PropTypes.func.isRequired,
  /** content header has editable title */
  editable: PropTypes.bool.isRequired,
  /** content header title */
  title: PropTypes.string.isRequired,
};

// EXPORT
export default ContentHeader;
