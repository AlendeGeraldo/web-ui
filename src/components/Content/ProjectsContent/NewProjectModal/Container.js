// CORE LIBS
import React from 'react';
import { connect } from 'react-redux';

// ACTIONS
import { createProject } from '../../../../store/projects/actions';

// COMPONENTS
import NewProjectModal from './index';

// MOCKS
// project mock
const projectMock = {
  createdAt: 'October 13, 2014 11:13:00',
};

// DISPATCHS
const mapDispatchToProps = (dispatch) => {
  return {
    handleCreateProject: (projectName) =>
      dispatch(
        createProject({
          ...projectMock,
          name: projectName,
          uuid: projectName.toLowerCase(),
        })
      ),
  };
};

/**
 * New Project Modal Container.
 * This component is responsible for create a logic container for new project
 * modal with redux.
 */
const NewProjectModalContainer = ({
  visible,
  handleCloseModal,
  handleCreateProject,
}) => (
  <NewProjectModal
    visible={visible}
    handleCloseModal={handleCloseModal}
    handleNewProject={handleCreateProject}
  />
);

// EXPORT
export default connect(null, mapDispatchToProps)(NewProjectModalContainer);
