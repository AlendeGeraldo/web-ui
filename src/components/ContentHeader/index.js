import './style.scss';
import React from 'react';
import { PageHeader } from 'antd';
// import PropTypes from 'prop-types';

const ContentHeader = ({
  title,
  onTitleDoubleClick,
  subTitle,
  onBack,
  breadcrumbs,
}) => {
  const componentTitle = onTitleDoubleClick ? (
    <span onDoubleClick={onTitleDoubleClick}>{title}</span>
  ) : (
    title
  );

  const routes = breadcrumbs;

  return (
    <PageHeader
      className='contentHeader'
      title={componentTitle}
      onBack={onBack}
      breadcrumb={{ routes }}
    />
  );
};

// ContentHeader.propTypes = {
//   title: PropTypes.string,
//   subTitle: PropTypes.string,
// };

// ContentHeader.defaultProps = {
//   title: '',
//   subTitle: '',
// };

export default ContentHeader;
