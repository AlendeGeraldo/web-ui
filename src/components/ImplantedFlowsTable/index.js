/**
 * Component responsible for:
 * - Structuring the table layout
 * - List all implantations
 */
import './style.scss';
import React from 'react';
import { Table, Typography, Tooltip } from 'antd';

const { Paragraph } = Typography;

const tableColumns = [
  {
    title: 'Nome',
    dataIndex: 'flowName',
    key: 'flowName',
  },
  {
    title: 'URL',
    dataIndex: 'url',
    key: 'url',
    render: (value) => (
      <Tooltip title={value}>
        <Paragraph ellipsis copyable>
          {value}
        </Paragraph>
      </Tooltip>
    ),
  },
  {
    title: 'Data de Criação',
    dataIndex: 'created',
    key: 'created',
  },
  {
    title: 'Ação',
    dataIndex: 'action',
    key: 'action',
    render: (value) => (
      <a target='_blank' href='/notebook/kubeflow-anonymous/server-1/tree?'>
        Monitoramento
      </a>
    ),
  },
];

const ImplantedFlowsTable = ({ flowList }) => {
  return (
    <Table
      dataSource={flowList}
      columns={tableColumns}
      pagination={{ pageSize: 9 }}
    />
  );
};

export default ImplantedFlowsTable;
