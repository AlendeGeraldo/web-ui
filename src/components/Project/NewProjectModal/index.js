// FIXME: tecla enter não está enviando formulário
/**
 * Component responsible for:
 * - Structuring the add project form layout
 * - Add new project
 */

import React from 'react';

import { Modal, Form, Input } from 'antd';
import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom';

import {
  addProject,
  toggleModal,
} from '../../../store/actions/projectsActions';

const NewProjectModal = (props) => {
  const { modalIsVisible, form } = props;
  const { onAddProject, onToggleModal } = props;
  const { getFieldDecorator, getFieldsError } = form;
  const history = useHistory();

  /**
   * Function used to check if form has errors
   * @param {string[]} fieldsError
   */
  const hasErrors = (fieldsError) => {
    return Object.keys(fieldsError).some((field) => fieldsError[field]);
  };

  /**
   * Function to handle modal cancel
   */
  const handleCancel = () => {
    form.resetFields();
    onToggleModal();
  };

  /**
   * Function to handle form submit
   * @param {Event} e
   */
  const handleSubmit = (e) => {
    e.preventDefault();
    form.validateFields(async (err, values) => {
      if (err) {
        return;
      }
      onAddProject(values.name, history);
    });
  };

  return (
    <Modal
      visible={modalIsVisible}
      title='Novo Projeto'
      okText='Criar'
      cancelText='Cancelar'
      onCancel={handleCancel}
      onOk={handleSubmit}
      okButtonProps={{ disabled: hasErrors(getFieldsError()) }}
    >
      <Form layout='vertical'>
        <Form.Item label='Qual o nome do seu projeto?'>
          {getFieldDecorator('name', {
            rules: [
              {
                required: true,
                message: 'Por favor insira um nome para o projeto!',
              },
            ],
            initialValue: 'Novo Projeto',
          })(<Input allowClear autoFocus />)}
        </Form.Item>
      </Form>
    </Modal>
  );
};

const mapStateToProps = (state) => {
  return {
    modalIsVisible: state.projects.modalIsVisible,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onAddProject: (name, history) => {
      dispatch(addProject(name, history));
    },
    onToggleModal: () => {
      dispatch(toggleModal());
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Form.create({ name: 'new-project' })(NewProjectModal));
