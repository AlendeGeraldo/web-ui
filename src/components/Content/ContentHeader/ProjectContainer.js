// CORE LIBS
import React, { useLayoutEffect } from 'react';
import { connect } from 'react-redux';
import { useHistory, useParams } from 'react-router-dom';

// COMPONENTS
import ContentHeader from './index';

// ACTIONS
import { fetchProject, editProjectName } from '../../../store/project/actions';

// DISPATCHS
const mapDispatchToProps = (dispatch) => {
  return {
    handleFetchProject: (projectUuid) => dispatch(fetchProject(projectUuid)),
    handleEditProjectName: (newName) =>
      dispatch(editProjectName(null, newName)),
  };
};

// STATES
const mapStateToProps = (state) => {
  return { project: state.project };
};

/**
 * Content Header Project Container.
 * This component is responsible for create a logic container for project content
 * header with route control.
 */
const ContentHeaderProjectContainer = ({
  project,
  handleFetchProject,
  handleEditProjectName,
}) => {
  // CONSTANTS
  // getting history
  const history = useHistory();
  // getting project uuid
  const { projectUuid } = useParams();

  console.log(projectUuid);

  // HOOKS
  // did mount hook
  useLayoutEffect(() => {
    // fetching projects
    handleFetchProject(projectUuid);
  }, []);

  // HANDLERS
  // go back
  const goBackHandler = () => history.goBack();

  // RENDER
  return (
    <ContentHeader
      title={project.name}
      editable
      handleGoBack={goBackHandler}
      handleSubmit={handleEditProjectName}
    />
  );
};

// EXPORT
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ContentHeaderProjectContainer);
