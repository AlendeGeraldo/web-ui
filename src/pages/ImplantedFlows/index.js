/**
 * Component responsible for:
 * - Structuring the implanted flows layout
 * - Fetch and List the implantations
 */
import './style.scss';
import React, { useState, useEffect } from 'react';
import { Empty, Spin } from 'antd';
import ContentHeader from '../../components/Content/ContentHeader';
import ImplantedFlowsTable from '../../components/ImplantedFlowsTable';
import emptyPlaceholder from '../../assets/emptyPlaceholder.png';
import { getDeployments } from '../../services/pipelinesApi';

const ImplantedFlows = (props) => {
  const [loading, setLoading] = useState(false);
  const [flowList, setFlowList] = useState([]);

  /**
   * Function to fetch the implatantions
   */
  const deploymentsFetch = async () => {
    setLoading(true);
    const response = await getDeployments();
    setLoading(false);
    if (response) setFlowList(response);
  };

  useEffect(() => {
    deploymentsFetch();
  }, []);

  /**
   * Funtion to render page body
   */
  const renderBody = () => {
    if (loading) return <Spin />;

    return flowList.length === 0 ? (
      <Empty
        image={emptyPlaceholder}
        imageStyle={{
          height: 136,
        }}
        description={
          <span>
            <span>
              <strong>Nenhum fluxo implantado</strong>
            </span>
            <br />
            <span>
              Crie fluxos em &quot;Projetos&quot; para depois implant√°-los
            </span>
          </span>
        }
      />
    ) : (
      <ImplantedFlowsTable flowList={flowList} />
    );
  };

  return (
    <div className='implantedFlowsPage'>
      <ContentHeader
        title='Fluxos implantados'
        subTitle='Monitore o desempenho do seu fluxo em tempo real.'
      />

      <div className='implantedFlowsPageBody'>
        <div className='body'>{renderBody()}</div>
      </div>
    </div>
  );
};

export default ImplantedFlows;
