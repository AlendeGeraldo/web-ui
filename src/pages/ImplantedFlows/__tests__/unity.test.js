import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';
import ContentHeader from '../../../components/ContentHeader';
import ImplantedFlowsTable from '../../../components/ImplantedFlowsTable';
import NewProjectModal from '../../../components/Project/NewProjectModal';
import ImplantedFlows from '..';

const implatedFlows = [
  {
    key: 'key',
    flowName: 'flowName',
    url: 'url',
    created: 'created',
    action: 'action',
  },
];

describe('ImplantedFlows component', () => {
  it('is expected render without crashing', () => {
    shallow(<ImplantedFlows />);
  });

  it('is expected render body with empty html correctly', () => {
    act(() => {
      const wrapper = mount(<ImplantedFlows />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
