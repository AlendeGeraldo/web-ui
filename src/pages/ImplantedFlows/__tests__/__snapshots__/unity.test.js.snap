// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImplantedFlows component is expected render body with empty html correctly 1`] = `
<ImplantedFlows>
  <div
    className="implantedFlowsPage"
  >
    <ContentHeader
      subTitle="Monitore o desempenho do seu fluxo em tempo real."
      title="Fluxos implantados"
    >
      <PageHeader
        backIcon={
          <Icon
            type="arrow-left"
          />
        }
        breadcrumb={
          Object {
            "routes": undefined,
          }
        }
        className="contentHeader"
        title="Fluxos implantados"
      >
        <div
          className="ant-page-header contentHeader ant-page-header-ghost"
        >
          <div
            className="ant-page-header-heading"
          >
            <span
              className="ant-page-header-heading-title"
            >
              Fluxos implantados
            </span>
          </div>
        </div>
      </PageHeader>
    </ContentHeader>
    <div
      className="implantedFlowsPageBody"
    >
      <div
        className="body"
      >
        <Empty
          description={
            <span>
              <span>
                <strong>
                  Nenhum fluxo implantado
                </strong>
              </span>
              <br />
              <span>
                Crie fluxos em "Projetos" para depois implantá-los
              </span>
            </span>
          }
          image="emptyPlaceholder.png"
          imageStyle={
            Object {
              "height": 136,
            }
          }
        >
          <LocaleReceiver
            componentName="Empty"
          >
            <div
              className="ant-empty"
            >
              <div
                className="ant-empty-image"
                style={
                  Object {
                    "height": 136,
                  }
                }
              >
                <img
                  alt="empty"
                  src="emptyPlaceholder.png"
                />
              </div>
              <p
                className="ant-empty-description"
              >
                <span>
                  <span>
                    <strong>
                      Nenhum fluxo implantado
                    </strong>
                  </span>
                  <br />
                  <span>
                    Crie fluxos em "Projetos" para depois implantá-los
                  </span>
                </span>
              </p>
            </div>
          </LocaleReceiver>
        </Empty>
      </div>
    </div>
  </div>
</ImplantedFlows>
`;
